### PR TITLE
 Avoid remote calls to get conversation when it is not found locally

### PR DESCRIPTION
**Pull request by akshaymankar**
Depends on #1703
Re-work of #1713 

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If end-points have been added or changed: the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [x] If a schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] Section *Unreleased* of **CHANGELOG-draft.md** contains the following bits of information:
   - [x] A line with the title and number of the PR in one or more suitable sub-sections.
   - [x] If /a: measures to be taken by instance operators.
   - [x] If /a: list of cassandra migrations.
   - [x] If public end-points have been changed or added: does nginz need upgrade?
   - [x] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

File: services/galley/src/Galley/API/Query.hs [hunk 1]
```diff
@@ -101,15 +101,18 @@ getConversation zusr cnv = do
...
-  conversations <- getRemoteConversations zusr [remoteConvId]
-  case conversations of
-    [] -> throwErrorDescription convNotFound
-    [conv] -> pure conv
-    _convs -> throwM (federationUnexpectedBody "expected one conversation, got multiple")
+    else getRemoteConversation (toRemote cnv)
+  where
+    getRemoteConversation :: Remote ConvId -> Galley Public.Conversation
+    getRemoteConversation remoteConvId = do
+      (foundConvs, _) <- Data.remoteConversationIdOf zusr [remoteConvId]
```
Comment (pcapriotti) [inline services/galley/src/Galley/API/Query.hs:108]: There is a compile error here. Should it just be `foundConvs <-`?

**Review by pcapriotti — COMMENTED** [review]
Looks good, but there is a compile error.

File: services/galley/src/Galley/API/Query.hs (hunk 1 above)
Comment (akshaymankar) [reply services/galley/src/Galley/API/Query.hs:108]: Ah yes, I refactored this. Note to self: it is seldom worth it to work on top of unmerged PR.

**Review by pcapriotti — APPROVED** [review]
:rocket:

**Pull request merged**